plugins/services/src/js/reducers/serviceForm/JSONReducers/Volumes.ts

```diff
@@ -197,11 +209,17 @@ export function JSONParser(state) {
         }
         setVolumeProp("size", size);
       } else if (item.external != null) {
-        setVolumeProp("type", "EXTERNAL");
-        setVolumeProp("name", item.external.name);
-        setVolumeProp("size", item.external.size);
-        setVolumeProp("options", item.external.options);
-        setVolumeProp("provider", item.external.provider);
+        if (item.external.provider.toLowerCase() === "csi") {
+          const data = { ...item, externalCSI: item.external };
+          memo.push(new Transaction(["volumes", index], data));
+          setVolumeProp("type", "EXTERNAL_CSI");
```

Comment:
the 4 lines above are a trick to not interfere with the existing DVDI-external volumes, as handling those in the same path is REALLY tricky. the internal model will store CSI-information at `externalCSI`. we unpack that in line `42`. obviously we have both, the `Parser` and the `JSONReducer` in this file... 🙈